subworkflows/local/input_check.nf

```diff
@@ -2,62 +2,39 @@
 // Check input samplesheet and get read channels
 //
 
-include { EIDO_VALIDATE } from '../../modules/nf-core/eido/validate/main'
-include { EIDO_CONVERT } from '../../modules/nf-core/eido/convert/main'
+include { SAMPLESHEET_CHECK } from '../../modules/local/samplesheet_check'
 
 workflow INPUT_CHECK {
     take:
-    samplesheet_or_pep_config // file: /path/to/samplesheet.csv or /path/to/pep/config.yaml
-    pep_input_base_dir
+    samplesheet // file: /path/to/samplesheet.csv
 
     main:
-    ch_versions = Channel.empty()
-
-    EIDO_VALIDATE ( samplesheet_or_pep_config, file("$projectDir/assets/samplesheet_schema.yaml"), pep_input_base_dir )
-    ch_versions = ch_versions.mix(EIDO_VALIDATE.out.versions)
-
-    EIDO_CONVERT ( samplesheet_or_pep_config, "csv", pep_input_base_dir )
-    ch_versions = ch_versions.mix(EIDO_CONVERT.out.versions)
-
-    ch_parsed_samplesheet = EIDO_CONVERT.out.samplesheet_converted
+    parsed_samplesheet = SAMPLESHEET_CHECK ( samplesheet )
+        .csv
         .splitCsv ( header:true, sep:',' )
-        .map { check_missing_and_singleend_autodetect(it) }
         .branch {
             fasta: it['fasta'] != ''
             nanopore: it['instrument_platform'] == 'OXFORD_NANOPORE'
             fastq: true
         }
 
-    ch_parsed_samplesheet.fastq
+    parsed_samplesheet.fastq
         .map { create_fastq_channel(it) }
         .set { fastq }
 
-    ch_parsed_samplesheet.nanopore
+    parsed_samplesheet.nanopore
         .map { create_fastq_channel(it) }
         .set { nanopore }
 
-    ch_parsed_samplesheet.fasta
+    parsed_samplesheet.fasta
```

Comment:
I prefer assignment to using `.set` but either is valid code, of course.

Reply:
Agreed! I have a feeing this is very old what I'm putting back in :sweat_smile: